content/en/blog/signal-boosting.md

```diff
@@ -0,0 +1,290 @@
+
+It should be noted that none of these concepts are invented by us. Reactive programming has quite a history, and has already been popularized widely in the JavaScript world by [Vue.js](https://vuejs.org/), [Svelte](https://svelte.dev/), [SolidJS](https://www.solidjs.com/), [RxJS](https://rxjs.dev/) and too many others to name. Kudos to all of them!
+
+
+# A Whirlwind Tour of the Signals Core
+
+Let's start with an overview of the fundamental features in the **@preact/signals-core** package.
+
+The code snippets below use functions imported from the package. The import statements are shown only when a new function is brought into the mix.
+
+### Signals
+
+Plain _signals_ are the fundamental root values which our reactive system is based on. Other libraries might call them, for example, "observables" ([MobX](https://mobx.js.org/observable-state.html), [RxJS](https://rxjs.dev/guide/observable)) or "refs" ([Vue](https://vuejs.org/guide/extras/reactivity-in-depth.html#how-reactivity-works-in-vue)). The Preact team adopted the term "signal" used by [SolidJS](https://www.solidjs.com/tutorial/introduction_signals).
+
+Signals represent arbitrary JavaScript values wrapped into a reactive shell. You provide a signal with an initial value, and can later read and update it as you go.
+
+```ts
+import { signal } from "@preact/signals-core";
+
+const s = signal(0);
+console.log(s.value); // Console: 0
+
+s.value = 1;
+console.log(s.value); // Console: 1
+```
+
+By themselves signals are not terribly interesting until combined with the two other primitives, _computed signals_ and _effects_.
+
+### Computed Signals
+
+_Computed signals_ derive new values from other signals using _compute functions_.
+
+```ts
+import { signal, computed } from "@preact/signals-core";
+
+const s1 = signal("Hello");
+const s2 = signal("World");
+
+const c = computed(() => {
+  return s1.value + " " + s2.value;
+});
+```
+
+The compute function given to `computed(...)` won't run immediately, so the `"I'm computing!` text won't get printed to the console. That's because computed signals are evaluated _lazily_, i.e. when their values are read.
+
+```ts
+console.log(c.value); // Console: Hello World
+```
+
+Computed values are also _cached_. A compute function can potentially be expensive, so we want to recompute only when it matters. A running compute function tracks which signal values are actually accessed during its run. When none of the signals have changed, then we can skip recomputation. In the above example, we can just reuse the previously calculated `c.value` indefinitely until either `a` or `b` changes. After that the compute function needs to be evaluated again on next read. Facilitating this _dependency tracking_ is the reason why we need to wrap the primitive values into signals in the first place.
```

Comment:
Don't know if it's worth it but we could couple back to a smart `memoize` function that knows when to recalculate

Reply:
Great suggestion 👍 Mentioning `memoize` would totally give some context for many readers. I tried modifying the text based on your suggestion, but couldn't figure out a way smooth way to do it yet.